rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp

```diff
@@ -346,6 +413,75 @@ TEST_F(RecordFixture, record_end_to_end_with_splitting_splits_bagfile) {
   }
 }
 
+TEST_F(RecordFixture, record_end_to_end_test_with_zstd_file_compression_compresses_files) {
+  constexpr const char topic_name[] = "/test_topic";
+  constexpr const int bagfile_split_size = 4 * 1024 * 1024;  // 4MB.
+
+  std::stringstream command;
+  command << "ros2 bag record" <<
+    " --output " << root_bag_path_ <<
+    " --max-bag-size " << bagfile_split_size <<
+    " --compression-mode file" <<
+    " --compression-format zstd"
+    " " << topic_name;
+
+  auto process_handle = start_execution(command.str());
+  wait_for_db();
+
+  constexpr const int expected_splits = 4;
+  {
+    constexpr const char message_str[] = "Test";
+    constexpr const int message_size = 1024 * 1024;  // 1MB
+    // string message from test_msgs
+    const auto message = create_string_message(message_str, message_size);
+    constexpr const int message_count = bagfile_split_size * expected_splits / message_size;
+
+    pub_man_.run_scoped_publisher(
+      topic_name,
+      message,
+      50ms,
+      message_count);
+  }
+
+  stop_execution(process_handle);
+
+  rosbag2_storage::MetadataIo metadata_io;
+
+  // TODO(zmichaels11): Remove when stop_execution properly SIGINT on Windows.
+  // This is required since stop_execution hard kills the proces on Windows,
+  // which prevents the metadata from being written.
```

Comment:
Does this mean this feature is broken on Windows? Did we test this on this platform?